_inc/lib/class.core-rest-api-endpoints.php

```diff
@@ -279,7 +279,9 @@ public static function route( $path, $classname, $method,
 	 */
 	public static function dismiss_notice( $data ) {
 		$notice = $data['notice'];
-		if ( isset( $notice ) && ! empty( $notice ) ) {
+		$param = $data->get_json_params();
+
+		if ( isset( $param['dismissed'] ) && $param['dismissed'] === true && isset( $notice ) && ! empty( $notice ) ) {
```

Comment:
Let's not nest too many lines of code, what if we just return an error here if otherwise is true? Also, could you please fold the long line?